api-client/src/main/java/com/takipi/api/core/url/UrlClient.java

```diff
@@ -6,34 +6,92 @@
 import java.io.Reader;
 import java.net.HttpURLConnection;
 import java.net.URL;
+import java.util.List;
 
 import org.apache.commons.io.IOUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
 import com.takipi.api.core.consts.ApiConstants;
 import com.takipi.common.util.Pair;
 
 public abstract class UrlClient {
+	
+	public enum Operation {GET, DELETE, POST, PUT } 
+	
+	public interface UrlClientObserver {
```

Comment:
I'd prefer this being in a separate class file, given this is a public interface.
The enum should go along with it :)